src/lib/core/components/visualizations/implementations/MapVisualization.tsx

```diff
@@ -461,25 +491,42 @@ function MapViz(props: VisualizationProps) {
       : undefined;
   }, [overlayResponse]);
 
-  // `vocabulary` is taken from the response, not the study metadata
-  // this is because not all values are shown (up to 8 or 7 most popular + 'other')
-  const vocabulary = useMemo(() => {
-    const rankedValues =
-      overlayResponse.value != null
-        ? overlayResponse.value.mapMarkers.config.rankedValues
-        : undefined;
-    return xAxisVariable?.type === 'string'
-      ? rankedValues
-      : // quick and dirty hack to sort by the bin-start in the string like "(625,750]"
-        sortBy(rankedValues, (value) => Number(value.split(/[[(,]/)[1]));
-  }, [overlayResponse.value, xAxisVariable]);
+  // If it's a string variable and a small vocabulary, use it as-is from the study metadata.
+  // This ensures that for low cardinality categoricals, the colours are always the same.
+  // Otherwise use the overlayValues from the back end (which are either bins or a Top7+Other)
+  const vocabulary =
+    xAxisVariable?.type === 'string' &&
+    xAxisVariable?.vocabulary != null &&
+    xAxisVariable.vocabulary.length <= 8
+      ? xAxisVariable.vocabulary
+      : overlayResponse.value?.mapMarkers.config.overlayValues;
+
+  /**
+   * Reset checkedLegendItems to all-checked (actually none checked)
+   * if ANY of the checked items are NOT in the vocabulary
+   * OR if ALL of the checked items ARE in the vocabulary
+   *
+   * TO DO: generalise this for use in other visualizations
+   */
+  useEffect(() => {
+    if (vizConfig.checkedLegendItems == null || vocabulary == null) return;
+    // could possibly optimize by also returning if vocabulary.length <= 8 ?
```

Comment:
this makes sense to me, since we would always see all legend items if `vocabulary.length <= 8`